.github/workflows/changes.yml

```
@@ -198,181 +173,62 @@ jobs:
-            - "src/internal_events/redis.rs"
-            - "src/sources/redis/**"
-            - "src/sources/util/**"
-            - "src/sinks/redis.rs"
-            - "src/sinks/util/**"
-            - "scripts/integration/redis/**"
-          splunk:
-            - "src/internal_events/splunk_hec.rs"
-            - "src/sources/splunk_hec/**"
-            - "src/sources/util/**"
-            - "src/sinks/splunk_hec/**"
-            - "src/sinks/util/**"
-            - "scripts/integration/splunk/**"
-          webhdfs:
-            - "src/sinks/webhdfs/**"
-            - "src/sinks/util/**"
-            - "scripts/integration/webhdfs/**"
 
+  # Detects changes that are specific to integration tests
+  int_tests:
+    runs-on: ubuntu-latest
+    if: ${{ inputs.int_tests }}
+    outputs:
+      amqp: ${{ steps.filter.outputs.amqp }}
+      appsignal: ${{ steps.filter.outputs.appsignal}}
+      aws: ${{ steps.filter.outputs.aws }}
+      axiom: ${{ steps.filter.outputs.axiom }}
+      azure: ${{ steps.filter.outputs.azure }}
+      clickhouse: ${{ steps.filter.outputs.clickhouse }}
+      databend: ${{ steps.filter.outputs.databend }}
+      datadog: ${{ steps.filter.outputs.datadog }}
+      dnstap: ${{ steps.filter.outputs.dnstap }}
+      docker-logs: ${{ steps.filter.outputs.docker-logs }}
+      elasticsearch: ${{ steps.filter.outputs.elasticsearch }}
+      eventstoredb: ${{ steps.filter.outputs.eventstoredb }}
+      fluent: ${{ steps.filter.outputs.fluent }}
+      gcp: ${{ steps.filter.outputs.gcp }}
+      humio: ${{ steps.filter.outputs.humio }}
+      http-client: ${{ steps.filter.outputs.http-client }}
+      influxdb: ${{ steps.filter.outputs.influxdb }}
+      kafka: ${{ steps.filter.outputs.kafka }}
+      logstash: ${{ steps.filter.outputs.logstash }}
+      loki: ${{ steps.filter.outputs.loki }}
+      mongodb: ${{ steps.filter.outputs.mongodb }}
+      nats: ${{ steps.filter.outputs.nats }}
+      nginx: ${{ steps.filter.outputs.nginx }}
+      opentelemetry: ${{ steps.filter.outputs.opentelemetry }}
+      postgres: ${{ steps.filter.outputs.postgres }}
+      prometheus: ${{ steps.filter.outputs.prometheus }}
+      pulsar: ${{ steps.filter.outputs.pulsar }}
+      redis: ${{ steps.filter.outputs.redis }}
+      splunk: ${{ steps.filter.outputs.splunk }}
+      webhdfs: ${{ steps.filter.outputs.webhdfs }}
+    steps:
+      - uses: actions/checkout@v3
+
+      # creates a yaml file that contains the filters for each integration,
+      # extracted from the output of the `vdev int paths` command, which
+      # sources the paths from the scripts/integration/.../test.yaml files
+      - name: Create filter rules for integrations
+        run: |
+          echo "all-int:" > int_test_filters.yaml
+          echo '- "lib/vector-core/**"' >> int_test_filters.yaml
+
+          cargo vdev int paths >> int_test_filters.yaml
+
+          echo "int_test_filters.yaml="
+          cat int_test_filters.yaml
+          echo ""
```

IMO, even more of this could go into `vdev`, possibly reducing this step down to just this:
```suggestion
          cargo vdev int ci-paths
```
Now, I don't think it _has to_ go into `vdev`, I just question the value of splitting the logic part within `vdev` and part in the CI workflow file.

---

I had considered this, but I elected to omit the vector-core filter, because technically that isn't an integration so it seemed a bit out of place.

---

I think I need to just accept that you are right most of the time LOL

Reflected on that more and, even though the lib vector-core isn't an integration, the purpose of the subcommand is to provide filters for what integration tests to run, in the case of vector-core that is all.

And since the subcommand name is "ci-paths", that eases my concerns about it not being too helpful to local users.